terraform/sgtm-cluster/main.tf

```diff
@@ -407,6 +408,43 @@ resource "aws_lambda_permission" "lambda_permission_for_sgtm_rest_api" {
   source_arn = "${var.sgtm_rest_api_execution_arn}/*/${aws_api_gateway_method.sgtm_post.http_method}${aws_api_gateway_resource.sgtm_resource.path}"
 }
 
+## SQS
+resource "aws_sqs_queue" "sgtm-webhooks-queue-fifo" {
+  name = "sgtm-webhooks-queue${local.suffix}.fifo"
+  fifo_queue = true
+  content_based_deduplication = true
+  visibility_timeout_seconds = 720  # 12 minutes
```

Comment:
SGTM's lambda timeout is set for 2 minutes right? Wouldn't we want to only lock the message for ~2 minutes with the assumption that if it hasn't been deleted yet then the lambda timed out or something went wrong? 12 minutes is a long time.

Reply:
ah I was following a recommendation from https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-queueconfig
<img width="777" alt="image" src="https://github.com/Asana/SGTM/assets/34071587/0f58a5c3-1032-423e-aeee-83ccfc6f0446">

The recommendation is for using the SQS built-in retry though, which we aren't doing here so seems reasonable to shorten it. We could consider using the SQS retry if we want to enable multiple retries for each webhook event.

Reply:
hmm yeah I'm not sure how applicable this is to us. The main concern around decreasing the visibility timeout is increasing the number of lockable events for dynamodb and increasing the rate of lock timeouts whereas the concern for the reverse would be slowing down SGTM's responsiveness in cases where the lambda does error/timeout. 🤔 I'll go with whatever you recommend.

Reply:
hm I'll bump it down to 4 minutes so there's a little bit of extra time, and I wouldn't expect things to be sitting in queue for much longer